src/tools/imgtool/library.h

```diff
@@ -81,20 +81,59 @@ struct imgtool_chainent
 	uint64_t block;
 };
 
-enum imgtool_forktype_t
+namespace imgtool
 {
-	FORK_END,
-	FORK_DATA,
-	FORK_RESOURCE,
-	FORK_ALTERNATE
-};
+	class fork_entry
```

Comment:
Dumb question - isn't that what `fork_entry(fork_entry &&that) = default;` does?